dot/sync/disjoint_block_set.go

```diff
@@ -80,13 +103,42 @@ type disjointBlockSet struct {
 
 	// map of parent hash -> child hashes
 	parentToChildren map[common.Hash]map[common.Hash]struct{}
+
+	timeNow func() time.Time
 }
 
 func newDisjointBlockSet(limit int) *disjointBlockSet {
 	return &disjointBlockSet{
 		blocks:           make(map[common.Hash]*pendingBlock),
 		parentToChildren: make(map[common.Hash]map[common.Hash]struct{}),
 		limit:            limit,
+		timeNow:          time.Now,
+	}
+}
+
+func (s *disjointBlockSet) start(ctx context.Context, done chan<- struct{}) {
+	defer close(done)
+	ticker := time.NewTicker(clearBlocksInterval)
```

Comment:
nit You can defer the ticker Stop here instead of having it in the `case <-ctx.Done():` block